### ISSUE TITLE
Bug: Pipeline validation is not working for default appsody project

### ISSUE THREAD

**Issue by mtamboli** [kabanero-io/kabanero-pipelines#220]
Pipeline validation is failing for projects created with `appsody init` using the 0.6 stack hub.

New design of stacks, adds the registry name in the .appsody.config file. Pipelines are failing even though the registry added is docker.io which is the default.
* Failure message is not useful. It needs to be improved. Failure message added below in steps to reproduce the problem.
```
{"level":"info","ts":1581897715.3239002,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
.appsody-config.yaml specifies stack kabanerobeta , but no versions of kabanerobeta are active, and cannot be built.
Step failed
```
My understanding is that pipeline validation should works as follows:

If no registry is specified for older projects stack: kabanerobeta/java-microprofile:0.2, assume docker.io and try to do validation
If registry name is specified stack: my-registry-url/kabanerobeta/java-microprofile:0.2, try to use that registry to see if that is active and pull the stack image from the defined registry assuming the secret is created for non-public registries.

**Comment by dacleyra** [comment]
@aadeshpa @smcclem 

This appears to be flawed
https://github.com/kabanero-io/kabanero-pipelines/blob/master/pipelines/incubator/validate-active-stack-task.yaml#L47

```
++ awk '{if ($1 ~ "stack:"){printf "STACK="$2}}' appsody-config.yaml
+ declare STACK=docker.io/kabanerobeta/java-microprofile:0.2
+ '[' -z docker.io/kabanerobeta/java-microprofile:0.2 ']'
++ echo docker.io/kabanerobeta/java-microprofile:0.2
++ cut -d: -f2
+ VERSION=0.2
++ echo docker.io/kabanerobeta/java-microprofile:0.2
++ cut -d: -f1
+ PROJECT_STACK=docker.io/kabanerobeta/java-microprofile
++ echo docker.io/kabanerobeta/java-microprofile
++ cut -d/ -f1
+ PROJECT=docker.io
++ echo docker.io/kabanerobeta/java-microprofile
++ cut -d/ -f2
+ STACK_NAME=kabanerobeta
```

**Comment by kvijai82** [comment]
@dacleyra @mtamboli have the first part of this fix done.  testing it out now.  will have a part 2 tomorrow that reconciles with a config map.

**Comment by kvijai82** [comment]
Fix for this in the 0.6.0-rc3 release.  There is a small follow up tweak coming in 0.6.0-rc4 today.

**Comment by marikaj123** [comment]
Need to add a fix to 0.6.0-rc.4
